docs/dev/tutorials/gated-nft-paymaster-tutorial.md

```diff
@@ -0,0 +1,734 @@
+# Gated NFT paymaster
+
+Discover how to construct a tailored paymaster that enables users to sidestep gas fees when they own a particular NFT. By the end of this tutorial, you'll:
+
+- Write a `mint` function for the `ERC721.sol` contract.
+- Design and implement the validation logic for the `ERC721GatedPaymaster.sol` contract.
+- Deploy `Greeter.sol`, `ERC721.sol`, and `ERC721GatedPaymaster.sol` contracts.
+- Connect our frontend application with users MetaMask wallets.
+- Initialize our contracts to facilitate interaction on the client side.
+- Fetch data to display in a React component.
+- Harness the capabilities of our paymaster based on NFT possession.
+
+### Introduction
+
+Welcome to this hands-on tutorial on creating and interacting with an `ERC721GatedPaymaster` and interacting with a front-end application. If you've ever been curious about the mechanisms that allow for gasless transactions on zkSync Era and our native account abstraction capabilities you're in the right place!
+
+You'll gain a thorough understanding of how to create, compile, and deploy smart contracts on zkSync Era, delve into the intricacies of paymasters and validation logic, and seamlessly integrate them within a frontend application. On completion, you'll have a fully functional `ERC721GatedPaymaster` contract and an interactive frontend to go with it.
+
+## Prerequisites
+
+- Make sure your machine satisfies the [system requirements](https://github.com/matter-labs/era-compiler-solidity/tree/main#system-requirements).
+- A [Node.js](https://nodejs.org/en/download) installation running Node.js version 16.
+- Some familiarity with deploying smart contracts on zkSync. If not, please refer to the first section of the [quickstart tutorial](../building-on-zksync/hello-world.md).
+- Some background knowledge on the concepts covered by the tutorial would be helpful too. Have a look at the following docs:
+  - [Account abstraction protocol](../../reference/concepts/account-abstraction.md).
+  - [Introduction to system contracts](../../reference/architecture/system-contracts.md).
+  - [Smart contract deployment](../../reference/architecture/contract-deployment.md) on zkSyn Era.
+  - [Gas estimation for transactions](../../reference/concepts/fee-model.md#gas-estimation-for-transactions) guide.
+- You should also know [how to get your private key from your MetaMask wallet](https://support.metamask.io/hc/en-us/articles/360015289632-How-to-export-an-account-s-private-key).
+
+### Complete project
+
+Download the complete project [here](https://github.com/matter-labs/gated-nft-tutorial).
```

Comment:
Should the completed project live within https://github.com/zkSync-Community-Hub/tutorials/tree/main/tutorials ?

Reply:
Defer to another PR